### DISCUSSION TITLE
I'm trying to fetch the data through Apollo Client in react native, the data is not getting fetched and giving errors, what should I do ?

### DISCUSSION THREAD

**Discussion by shubhamk-27** [vendurehq/vendure#1131] [Q&A]
import React from 'react';
import {Text, View, FlatList} from 'react-native';
import MI from 'react-native-vector-icons/MaterialIcons';
import {
  ApolloProvider,
  ApolloClient,
  InMemoryCache,
  useQuery,
  gql,
} from '@apollo/client';

const GET_ALL_PRODUCTS = gql`
  {
    products {
      items {
        name
      }
    }
  }
`;

const HomeScreen = props => {
  const data = useQuery(GET_ALL_PRODUCTS);
  JSON.parse(data);

  return (
    <View>
      <MI name="menu" size={30} onPress={() => props.navigation.openDrawer()} />
      <Text>Items</Text>
      <View
        style={{
          display: 'flex',
          flex: 1,
          alignItems: 'center',
          justifyContent: 'center',
        }}>
        <FlatList
          data={data.products}
          renderItem={({item}) => <Text>{item.name}</Text>}
        />
      </View>
    </View>
  );
};
export default HomeScreen;

**Comment by michaelbromley** [comment]
What errors? Please provide specifics.

**Reply by shubhamk-27** [reply]
I get this error, when I try to fetch data

![Screenshot_2021-10-09-14-51-22-15_bb410cfd332a9ea16c9b6c66ed0598bc.jpg](https://user-images.githubusercontent.com/72503268/136687940-7cf8ac02-548b-45cd-b772-c31302bfe116.jpg)

**Reply by michaelbromley** [reply]
Looks like maybe that `data` variable is not an array, so the `.map()` function is undefined. This is just something you need to debug - it is not related to Vendure in particular and I can't help any better than say, someone on StackOverflow.
